### PR TITLE
hashes: Make more effort with the macros docs

### PR DESCRIPTION
**Pull request by tcharding**
A quick cleanup to try and make the docs better around our macros.

Last patch might be churn, can drop it if wanted. Came about because the `macros` module shows empty in the docs currently.

### CODE REVIEW

**Comment by apoelstra** [comment]
Interesting about needing to link them to make them show up in the docs. Sounds like a bug in rustdoc to me. But this looks good to me. Will ack.

File: hashes/src/sha256t.rs [hunk 1]
```diff
@@ -211,7 +176,7 @@ macro_rules! sha256t_tag {
 ///
 /// [`hash_newtype`]: crate::hash_newtype
 #[macro_export]
-#[deprecated(since = "TBD", note = "use `sha256_tag!` combined with `hash_newtype!` instead")]
+#[deprecated(since = "0.15.0", note = "use `sha256_tag!` combined with `hash_newtype!` instead")]
```
Comment (jamillambert) [inline hashes/src/sha256t.rs:179]: Add `/// # #[allow(deprecated)] {` to the example code in `hashes/src/sha256t.rs:154` to remove the lint warning.

Comment (tcharding) [reply hashes/src/sha256t.rs:181]: Thanks man, I don't know why it passed CI. I created #3529

**Review by apoelstra — APPROVED** [review]
ACK 24c33dc56a8e3813935db0fd54679e023e869a19; successfully ran local tests

**Comment by tcharding** [comment]
Hey @jamillambert, as someone newer to the `bitcoin_hashes` crate, could you take a look at the docs with this patch applied and see if anything stands out to you as confusing or wrong? I mean to check the state of the docs after this PR not the changes in this PR.

**Review by apoelstra — APPROVED** [review]
ACK abcac540782dd84992e47f342929efd407746c65; successfully ran local tests

**Pull request merged**